lib/grpmat.gi

```diff
@@ -1241,6 +1241,146 @@ InstallGlobalFunction( "BlowUpIsomorphism", function( matgrp, B )
     end );
 
 
+#############################################################################
+##
+#F  DiagonalAutomorphismGroupOfSL(<G>)
+## 
+##  Returns the group of all diagonal automorphsims of G = SL(d,q)
+##  An automorphism f : G -> G is a diagonal automorphism iff
+##  f(U) = D U D^-1 for all U in G and some diagonal matrix D in GL(d,q)
+##
+InstallGlobalFunction("DiagonalAutomorphismGroupOfSL",function(G)
+    local D,x,z,i,M,d,q;
+    d := SLDegree(G);
+    q := Size(SLUnderlyingField(G));
+    z := Z(q);
+    D := [];
+    for i in [1..d] do
+        M := IdentityMatrix(GF(q),d);
+        M[i][i] := z;
+        if ConjugateGroup(G,M) = G then
```

Comment:
Why would you need this check? When could it fail if `G` is truly `SL_n` (which you of course currently don't verify, but you clearly assume it anyway)